stack2nix/Main.hs

```diff
@@ -39,9 +39,16 @@ args = Args
     -- | Copied from cabal2nix/src/Cabal2nix.hs
     platformReader :: P.ReadP r Platform
     platformReader = do
```

Comment:
Note that I the `platformReader` function is redundant. It can be replaced with [Distribution.Text.parse](http://hackage.haskell.org/package/Cabal-2.2.0.1/docs/Distribution-Text.html). I didn't do that because the haddocks say it may be deprecated;

```
-- | /Note:/ this class will soon be deprecated.
-- It's not yet, so that we are @-Wall@ clean.
class Text a where
```

If you want to try this change then either replace `platformReader` with `parse` or implement it like so;

```
import Distribution.Text (parse)

platformReader :: P.ReadP r Platform
platformReader = parse
```

Reply:
Since the code was copied from cabal2nix/src/Cabal2nix.hs it has been changed upstream to `maybeReader parsePlatform`;

https://github.com/NixOS/cabal2nix/blob/a77f6a79a574b61ebc631634264ac179361bf200/src/Cabal2nix.hs#L144-L145